### DISCUSSION TITLE
Fetch tests using before:spec or before:run

### DISCUSSION THREAD

**Discussion by JonahKK** [cypress-io/cypress#14890] [Questions and Help] [answered]
I would like to get the test (as `Mocha.Text`), before executing the spec.
I saw that recently Cypress added new events like `before:spec` and `before:run` yet after reading the documentation regarding those two events I was not able to get those. I want to check the test's name and depending on that, skip the test or run it.
Currently I'm using `Cypress.on('test:before:run', (_attributes, runnable)`, to achieve it, but this is quite slow solution, as the check is done before every test. Being able to do it before the run, or spec would make things much easier.

Maybe some has figured it out, cheers.

**Comment by bahmutov** [comment, marked as answer]
Unfortunately, this is not possible. When `before:spec` or `before:run` executes, the test names are unknown yet, thus you cannot decide if you can skip the spec or not.
